src/main/java/org/candlepin/subscriptions/db/model/HardwareMeasurementType.java

```diff
@@ -21,12 +21,32 @@
 
 package org.candlepin.subscriptions.db.model;
 
+import java.util.Arrays;
+import java.util.List;
+
 /**
  * Enum to capture the various types of measurements in the hardware_measurements table
  */
 public enum HardwareMeasurementType {
     PHYSICAL,
     HYPERVISOR,
-    TOTAL;
+    TOTAL,
+    AWS,
+    GOOGLE,
+    ALIBABA,
+    AZURE;
+
+    public static HardwareMeasurementType getCloudProvider(String name) {
+        for (HardwareMeasurementType cloudType : getCloudProviderTypes()) {
+            if (cloudType.name().equalsIgnoreCase(name)) {
+                return cloudType;
+            }
+        }
+        return null;
```

Comment:
Instead of returning null, I'd like to suggest that we have a method like `isSupportedCloudProvider` and the semantics would be like

```java
if (HardwareMeasurementType.isSupportedCloudProvider(normalizedFacts.getCloudProvider()) {
...
}
```

To me, this makes more sense than a comparison against null and it's also more succinct.  Plus, the implementation is just a `getCloudProviderTypes.contains(x)`

Reply:
DONE